runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/util/MemoryMonitorTest.java

```diff
@@ -137,9 +140,33 @@ public void uploadToGcs() throws Exception {
     assertThat(files[0].getAbsolutePath(), Matchers.containsString("hprof"));
   }
 
+  @Test
+  public void uploadJfrProfilesOnThrashing() throws Exception {
+    if (Environments.getJavaVersion() != Environments.JavaVersion.java8) {
```

Comment:
Use Assume#assumeFalse https://junit.org/junit4/javadoc/4.12/org/junit/Assume.html#assumeFalse(boolean) instead of an if.